src/pdm/cli/commands/run.py

```diff
@@ -216,12 +227,21 @@ def _run_task(
         kind, _, value, options = task
         shell = False
         if kind == "cmd":
-            if not isinstance(value, list):
-                value = shlex.split(str(value))
-            args = value + list(args)
+            interpolations = 0
+            if isinstance(value, str):
+                cmd, count = interpolate(value, args)
+                interpolations += count
+                value = shlex.split(cmd)
+            else:
+                agg = [interpolate(part, args) for part in value]
+                parts, counts = zip(*agg)
+                interpolations += sum(counts)
+                value = list(parts)
```

Comment:
Given the command: `["foo", "{args}", "bar"]` and `arg=--default --other`, I assume the value would be `["foo", "--default --other", "bar"]`, right?

Reply:
Hum, I guess you are right, I haven't tested the `cmd` as list case. I will

Reply:
Test cases updated and cases fixed